### PR TITLE
Guicing controller

### PR DESCRIPTION
**Pull request by eishay**
First pass with the bookmarks controller.
Its a great exercise, showed me how much concerns this controller has and some strange cases as using both URIGraph and URIGraphPlugin. It gives a good indications of what controllers we should break apart.

### CODE REVIEW

**Comment by gmethvin** [comment]
Nice!

We also need to set up FortyTwoController to get dependencies through constructor injection. The most obvious way is to make it an abstract class and pass dependencies through the constructor.

I guess we want to work toward organizing dependencies better so we don't have 20 constructor parameters on a controller.

**Comment by eishay** [comment]
agree. we should split the controller and use more delegation to classes
with focused concerns.
as a rule of thumb, if you have more then ten dependencies to a class then
something's wrong :-)

File: shoebox/app/com/keepit/controllers/BookmarksController.scala [hunk 1]
```diff
@@ -27,25 +26,33 @@ import scala.concurrent.duration._
 
 import com.keepit.common.analytics.ActivityStream
 
+import com.google.inject.{Inject, Singleton}
 
-object BookmarksController extends FortyTwoController {
+@Singleton
+class BookmarksController @Inject() (db: DBConnection,
```
Comment (2is10) [inline shoebox/app/com/keepit/controllers/BookmarksController.scala:32]: Maybe we should rename the `DBConnection` class. Injecting a database connection into a singleton object would normally be a big red flag.

Comment (eishay) [inline shoebox/app/com/keepit/controllers/BookmarksController.scala:32]: How would you name it?
Database, DB ?

**Comment by 2is10** [comment]
Looks good to me.

Yes, `BookmarksController` is ready to split. For now I'd like to propose `website` | `extension API` | `admin` as a highest level grouping of routes. Each controller should probably handle actions from only one of those categories.

**Comment by eishay** [comment]
I like the idea, I think Andrew mentioned something like this in the past. 
So should we have { FortyTwoAdminController, FortyTwoApiController and FortyTwoController } where you have AdminHtmlAction only within the FortyTwoAdminController ?
\cc @andrewconner @gmethvin

File: shoebox/app/com/keepit/controllers/BookmarksController.scala (hunk 1 above)
Comment (2is10) [inline shoebox/app/com/keepit/controllers/BookmarksController.scala:32]: Sure, `Database` would be fine, assuming we don't expect name conflicts to happen often. I don't know the Slick classes well, and I'm not very clear on what each of our additional database types/abstractions are for.

**Pull request merged**